dev/sg/sg_feedback.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/open"
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/std"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+const newDiscussionURL = "https://github.com/sourcegraph/sourcegraph/discussions/new"
+
+type stopReadFunc func(lastRead string, err error) bool
+
+func addFeedbackSubcommand(commands []*cli.Command) {
+	giveFeedback := false
+	feedbackFlag := cli.BoolFlag{
+		Name:        "feedback",
+		Usage:       "provide feedback about this command by opening up a Github discussion",
+		Destination: &giveFeedback,
+	}
+
+	for _, command := range commands {
+		command.Flags = append(command.Flags, &feedbackFlag)
+		action := command.Action
+		command.Action = func(ctx *cli.Context) error {
+			if giveFeedback {
+				return feedbackExec(ctx)
+			}
+
+			return action(ctx)
+		}
+
+		addFeedbackSubcommand(command.Subcommands)
+	}
+}
+
+var feedbackCommand = &cli.Command{
+	Name:     "feedback",
+	Usage:    "opens up a Github discussion page to provide feedback about sg",
+	Category: CategoryCompany,
+	Action:   feedbackExec,
+}
+
+func feedbackExec(ctx *cli.Context) error {
+	title, body, err := gatherFeedback(ctx)
+	if err != nil {
+		return err
+	}
+	body = addSGInformation(ctx, body)
+
+	if err := sendFeedback(ctx.Context, title, "developer-experience", body); err != nil {
+		return err
+	}
+	return nil
+}
+
+func gatherFeedback(ctx *cli.Context) (string, string, error) {
+	std.Out.WriteNoticef("Gathering feedback for sg %s", ctx.Command.FullName())
+
+	fmt.Println("What is the title of your feedback ?")
+	title, err := readUntilDelim(os.Stdin, '\n')
+
+	fmt.Println("Write your feedback below and press <CTRL+D> when you're done.")
+	body, err := readUntilEOF(os.Stdin)
```

Comment:
Sorry I chimed in only after you wrote all the code before, but "read until EOF" is by definition `io.ReadAll`:

```suggestion
	body, err := io.ReadAll(os.Stdin)
```

<img width="547" alt="image" src="https://user-images.githubusercontent.com/23356519/173904513-0a8da47f-f02a-46b9-ab0d-be9e3f3ceeeb.png">


<img width="780" alt="image" src="https://user-images.githubusercontent.com/23356519/173904549-30cab874-cf17-4639-87fa-c8814bfa4348.png">

Reply:
cc @jhchabran as well 😛

Reply:
nothing like reinventing the wheel 😆